common/environment/environment.cpp

```diff
@@ -1336,12 +1536,221 @@ void CLocalEnvironment::clearCache()
+    maxIndex = constEnv->getNumberOfMachines();
+    index = 1;
+    curr = NULL;
+    if (maxIndex)
+        curr = constEnv->getMachineByIndex(index);
+}
+
+bool CConstMachineInfoIterator::first()
+{
+    index = 1;
+    curr = constEnv->getMachineByIndex(index);
+    return (curr != NULL);
+}
+bool CConstMachineInfoIterator::next()
+{
+    if (index < maxIndex)
+    {
+        index++;
+        curr = constEnv->getMachineByIndex(index);
+    }
+    else
+        curr = NULL;
+
+    return (curr ? true : false);
+}
+
+bool CConstMachineInfoIterator::isValid()
+{
+    return curr != NULL;
+}
+
+IConstMachineInfo & CConstMachineInfoIterator::query()
+{
+    return *curr;
+}
+
+unsigned CConstMachineInfoIterator::count()
+{
+    return maxIndex;
+}
+
+//--------------------------------------------------
+
+CConstDropZoneIteratorByComputer::CConstDropZoneIteratorByComputer(const char * computer)
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
+    computerName.set(computer);
+    maxIndex = constEnv->getNumberOfDropZonesByComputer(computer);
+    index = 1;
+    curr = NULL;
+    if (maxIndex)
+        curr = constEnv->getDropZoneByComputerByIndex(computer, index);
+}
+
+bool CConstDropZoneIteratorByComputer::first()
+{
+    index = 1;
+    curr = constEnv->getDropZoneByComputerByIndex(computerName.str(), index);
+    return (curr != NULL);
+}
+bool CConstDropZoneIteratorByComputer::next()
+{
+    if (index < maxIndex)
+    {
+        index++;
+        curr = constEnv->getDropZoneByComputerByIndex(computerName.str(), index);
+    }
+    else
+        curr = NULL;
+
+    return (curr ? true : false);
+}
+
+bool CConstDropZoneIteratorByComputer::isValid()
+{
+    return curr != NULL;
+}
+
+IConstDropZoneInfo & CConstDropZoneIteratorByComputer::query()
+{
+    return *curr;
+}
+
+unsigned CConstDropZoneIteratorByComputer::count()
+{
+    return maxIndex;
+}
+
+//--------------------------------------------------
+
+CConstDropZoneInfoIterator::CConstDropZoneInfoIterator()
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
+    maxIndex = constEnv->getNumberOfDropZones();
+    index = 1;
+    curr = NULL;
+    if (maxIndex)
+        curr = constEnv->getDropZoneByIndex(index);
+}
+
+bool CConstDropZoneInfoIterator::first()
+{
+    index = 1;
+    curr = constEnv->getDropZoneByIndex(index);
+    return (curr != NULL);
+}
+
+bool CConstDropZoneInfoIterator::next()
+{
+    if (index < maxIndex)
+    {
+        index++;
+        curr = constEnv->getDropZoneByIndex(index);
+    }
+    else
+        curr = NULL;
+
+    return (curr ? true : false);
+}
+
+bool CConstDropZoneInfoIterator::isValid()
+{
+    return curr != NULL;
+}
+
+IConstDropZoneInfo & CConstDropZoneInfoIterator::query()
+{
+    return *curr;
+}
+
+unsigned CConstDropZoneInfoIterator::count()
```

Comment:
minor: could be const
The other IIterator methods should also be const really, but that's a pre-existing definition.